src/Microsoft.VisualStudio.LanguageServices.Razor/DefaultTagHelperResolver.cs

```diff
@@ -90,5 +105,11 @@ private void AddXmlDocumentation(Compilation compilation, IReadOnlyList<TagHelpe
                 }
             }
         }
+
+        private IVsActivityLog GetActivityLog()
+        {
+            var services = (IServiceProvider)Services;
```

Comment:
It's not strictly necessary in this case, but this is the general pattern. VS types have a 'service interface' which like a marker.